### DISCUSSION TITLE
Langchain integration Improvements

### DISCUSSION THREAD

**Discussion by ajram23** [langfuse/langfuse#1916] [Ideas]
### Describe the feature or potential improvement

Langsmith sends tags & session ID inside metadata. Would be great if you can extract those directly.

**Comment by marcklingen** [comment]
Great change, added as issue: https://github.com/langfuse/langfuse/issues/1918

**Reply by ajram23** [reply]
@marcklingen here is the relevant discussion in connection with tags as we chatted today. cc @marliessophie @hassiebp

**Reply by marcklingen** [reply]
thanks for the pointer here, we'll try to ad this soonish

**Comment by hassiebp** [comment]
Hi @ajram23 - this has been shipped in [v2.51.0](https://github.com/langfuse/langfuse-python/releases/tag/v2.51.0)! Please also check out [our docs](https://langfuse.com/docs/integrations/langchain/tracing#dynamic-trace-attributes-in-chain-invocation) on how to implement this 👍🏾

**Reply by ajram23** [reply]
@hassiebp thanks for shipping this! Will the tag be automatically pulled into the UI. Not seeing it. Here is the trace I ran https://us.cloud.langfuse.com/project/cltolaqjv0000m2pkhvdzv9f6/traces/88cf9f2d-ea01-40b9-8fb8-d7643b07326b

**Reply by hassiebp** [reply]
Yes, it is expected to be pulled automatically into the UI. Could you kindly share how you invocate your chain?🤔 I see the tags showing up correctly in the metadata. Just to confirm, you are on 2.51.0 of Langfuse Python, correct?

**Reply by ajram23** [reply]
@hassiebp Turned to be venv issue. I can confirm this works like a charm! well done! You sir are a shipping machine! Thank you!

**Reply by hassiebp** [reply]
Awesome, great to hear AJ! 🤩
